File: A_Treasure_Hunt.java
import java.util.*;

public class A_Treasure_Hunt {
    public static void solve(Scanner scanner) {
        long x = scanner.nextLong();
        long y = scanner.nextLong();
        long k = scanner.nextLong();

        if (x > k) {
            System.out.println("NO");
            return;
        }

        long s = x + y;
        long cycle = k / s;
        long remainder = k - (x * cycle + y * cycle);

        if (x > remainder) {
            System.out.println("NO");
        } else {
            System.out.println("YES");
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        while (t-- > 0) {
            solve(scanner);
        }
    }
}

File: A_Treasure_Hunt.cpp
#include <iostream>
using namespace std;

#define ll long long

void solve() {
    ll x, y, k;
    cin >> x >> y >> k;

    if (x > k) {
        cout << "NO" << endl;
        return;
    }

    ll s = x + y;
    ll cycle = k / s;
    ll remainder = k - (x * cycle + y * cycle);

    if (x > remainder) {
        cout << "NO" << endl;
    } else {
        cout << "YES" << endl;
    }
}

int main() {
    int t;
    cin >> t;
    while (t--) {
        solve();
    }
    return 0;
}
